src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Value.ReadTests.cs

```diff
@@ -442,5 +447,32 @@ private static void DeserializeLongJsonString(int stringLength)
             string str = JsonSerializer.Deserialize<string>(json);
             Assert.True(json.AsSpan(1, json.Length - 2).SequenceEqual(str.AsSpan()));
         }
+
+        [Theory]
+        [InlineData("23:59:59")]
+        [InlineData("23:59:59.9")]
+        [InlineData("23:59:59.9999999")]
+        [InlineData("9999999.23:59:59.9999999")]
+        [InlineData("-9999999.23:59:59.9999999")]
+        public static void TimeSpan_Read_Success(string json)
+        {
+            TimeSpan value = JsonSerializer.Deserialize<TimeSpan>($"\"{json}\"");
+
+            Assert.Equal(TimeSpan.Parse(json), value);
+        }
+
+        [Theory]
+        [InlineData("24:00:00")]
+        [InlineData("00:60:00")]
+        [InlineData("00:00:60")]
+        [InlineData("00:00:00.00000009")]
+        [InlineData("900000000.00:00:00")]
+        [InlineData("+00:00:00")]
+        [InlineData("2021-06-18")]
+        [InlineData("1$")]
```

Comment:
Please include boundary values such as the serializations of `TimeSpan.MaxValue + 1 tick` and `TimeSpan.MinValue - 1 tick`.

Reply:
Also, testing against non-string JSON values might be useful here.

Reply:
* Added the `TimeSpan.MaxValue` \ `TimeSpan.MinValue` cases.

* Added some non-string JSON value cases.